### PR TITLE
2.x sort locale backport

### PR DESCRIPTION
**Pull request by dereuromark**
Backport https://github.com/cakephp/cakephp/pull/9335 from 3.x to 2.x

### CODE REVIEW

File: lib/Cake/Test/Case/Utility/HashTest.php [hunk 1]
```diff
@@ -1361,6 +1361,38 @@ public function testSortNaturalFallbackToRegular() {
...
+ * Test sort() with locale option.
+ *
+ * @return void
+ */
+	public function testSortLocale() {
+		// get the current locale
+		$oldLocale = setlocale(LC_COLLATE, '0');
+
+		// the de_DE.utf8 locale must be installed on the system where the test is performed
+		setlocale(LC_COLLATE, 'de_DE.utf8');
```
Comment (renan) [inline lib/Cake/Test/Case/Utility/HashTest.php]: If the system doesn't have this locale the test will just fail, perhaps you could skip this test if `setlocale(LC_COLLATE, 'de_DE.utf8');` returns false.

From php.net/setlocale:
> Returns the new current locale, or FALSE if the locale functionality is not implemented on your platform, the specified locale does not exist or the category name is invalid.

Comment (dereuromark) [inline lib/Cake/Test/Case/Utility/HashTest.php]: Interesting. Are we setting it up differently in 3.x?

Comment (ADmad) [inline lib/Cake/Test/Case/Utility/HashTest.php]: @dereuromark Our 3.x testsuite runs on travis' "trusty" VMs which probably have the locale already.

Comment (dereuromark) [inline lib/Cake/Test/Case/Utility/HashTest.php]: Actually, see https://github.com/cakephp/cakephp/commit/8694e6b680cddefe3b17e41e0ee45b9288932ede on master.

**Comment by dereuromark** [comment]
The CakeTimeTest::testListTimezones fail is unrelated.

**Pull request merged**